code-challenges.js

```diff
@@ -8,62 +8,162 @@
+// const currentCohort = "Charlie 2023
 // Pseudo code:
+// console.log(currentCohort.reverse());--> wasnt that simple, should have known better
+
+// Thinking for some reason about creating a function, however I feel that is not needed and just a bunch of extra steps.  I thibk it could be done, but need to keep the code simple. 
+// .reverse is the way to go for sure. 
+// my variable contains the string, I can call my variable in console.log then split, reverse, join if I want this action to be temporary.  I will make it permenant by creating a variable called backwards and storing currentCohort split, reverse, join, then console.log
 
 const currentCohort = "Charlie 2023"
+
+const backwards = currentCohort.split("").reverse().join("")
+
+console.log(backwards)
 // Expected output: "3202 eilrahC"
+// // output is 3202 eilrahC
+
+
 
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
 
 // Pseudo code:
-
+// I know I need to find the index of the last 42,
+// i had to google research this.  the url is https://www.google.com/search?q=what+built+in+method+in+javascript+will+give+you+the+last+index+of+an+array&oq=what+built+in+method+in+javascript+will+give+you+the+last+index+of+an+array&aqs=chrome..69i57.16836j0j15&sourceid=chrome&ie=UTF-8  
+// I have an array, and a variable.  I will create a variable to be called on that will store the last indexed array and the givenValue1
+// the answer variable will call the array with the lastIndexOf built in method and pass in the arguement of givenValue1
+//                            0   1   2  3  4  5   6   7   8
 const numberOfConnections = [13, 34, 42, 5, 5, 10, 27, 42, 10]
 
 const givenValue1 = 42
-// Expected output: 7
+
+const answer1 = numberOfConnections.lastIndexOf(givenValue1)
+console.log(answer1);
+// output is 7
+// // Expected output: 7
+
+// repeated the same as above
 
 const givenValue2 = 10
+const answer2 = numberOfConnections.lastIndexOf(givenValue2)
+console.log(answer2);
+//output is 8
 // Expected output: 8
 
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
 // Pseudo code:
+// Blocker.  I am unsure how to code this, even after researching, using notes and rewatching lecture exams I had to resort to using chatGPT.  I understand you would have to use the .map and .sort built in methods but thst is about it.  Would like to possibly set up a time to go rework this problem
+
 
 const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
+
+const numbers = [5, 2, 8, 4, 1]
+
+const smallestToLargest = sanDiegoSummerTemperatures.map(function(temp) {
+  return temp;
+}).sort(function(a, b) {
+  return b - a;
+});
+
+const sortedNumbers = numbers.map(function(number) {
+  return number;
+}).sort(function(a, b) {
+  return b - a;
+});
+console.log(smallestToLargest);
```

Comment:
The `.map()` higher order function is a method that iterates through an array and takes a callback function as an argument. The way it is being implemented here, it is not actually doing anything! You would get the same result by using `.sort( (a,b) => b - a )` on each test variable.